### PR TITLE
Fixed comments, Class Casing, and misc code generation formatting

### LINKED ISSUE #1: delegate comments are generated in the wrong location
Current:

```C#
/// <summary>
/// The summary for the cpp function delegate
/// </summary>
/// <param name="parameter1">parameter description.</param>
/// <param name="parameter2">parameter description.</param>
/// <param name="parameter3">parameter description.</param>
/// <remarks>
/// @brief This callback will be invoked whenever event is detected.
/// </remarks>
public ClassName.Callbacks.function_delegate event;

[UnmanagedFunctionPointer(CallingConvention.Cdecl)]
public delegate void function_delegate (byte parameter1, in Interop.DeviceState parameter2, IntPtr parameter3);
```

Expected:

```C#
/// <summary>
/// The summary for the cpp function delegate event.
/// </summary>
public ClassName.Callbacks.function_delegate event;

/// <summary>
/// The summary for the cpp function delegate.
/// </summary>
/// <param name="parameter1">parameter description.</param>
/// <param name="parameter2">parameter description.</param>
/// <param name="parameter3">parameter description.</param>
/// <remarks>
/// This callback will be invoked whenever event is detected.
/// </remarks>
[UnmanagedFunctionPointer(CallingConvention.Cdecl)]
public delegate void function_delegate (byte parameter1, in Interop.DeviceState parameter2, IntPtr parameter3);
```

### PR DESCRIPTION
**Pull request by StephenHodgson**
- Fixes #1

### CODE REVIEW

File: src/CppAst.CodeGen/CSharp/Comments/CSharpReturnComment.cs [hunk 1]
```diff
@@ -2,17 +2,19 @@
...
 
 namespace CppAst.CodeGen.CSharp
 {
     public class CSharpReturnComment : CSharpComment
     {
+        /// <inheritdoc />
         public override void DumpTo(CodeWriter writer)
         {
-            writer.Write("<returns>");
+            writer.Write($"<returns>{Environment.NewLine}");
```
Comment (StephenHodgson) [inline src/CppAst.CodeGen/CSharp/Comments/CSharpReturnComment.cs]:
```suggestion
            writer.Write("<returns>\n");
```

File: src/CppAst.CodeGen/CSharp/Comments/CSharpReturnComment.cs [hunk 2]
```diff
@@ -2,17 +2,19 @@
...
             DumpChildrenTo(writer);
-            writer.WriteLine("</returns>");
+            writer.WriteLine($"{Environment.NewLine}</returns>");
```
Comment (StephenHodgson) [inline src/CppAst.CodeGen/CSharp/Comments/CSharpReturnComment.cs]:
```suggestion
            writer.WriteLine("\n</returns>");
```

**Comment by StephenHodgson** [comment]
This is still in progress

**Comment by StephenHodgson** [comment]
Blocked by #20

**Comment by StephenHodgson** [comment]
Blocked by #17

**Comment by Perksey** [comment]
What's the status on this?

**Comment by StephenHodgson** [comment]
I've been super busy and was able to get pretty close to the results I wanted and manually edited the output.

**Pull request closed without merging**